docs/getting-started/android-integration.md

```diff
@@ -0,0 +1,243 @@
+            // target your experiments.
+            put("is_first_run", isAppFirstRun)
+        }
+
+        val appInfo = NimbusAppInfo(
+            appName = "my-app-name",
+            channel = BuildConfig.BUILD_TYPE,
+            customTargetingAttributes = customTargetingAttributes
+        )
+
+        // Use the Nimbus builder to build a NimbusApi object.
+        return NimbusBuilder(context).apply {
+            url = urlString
+            errorReporter = { message, e ->
+                Logger.error("Nimbus error: $message", e)
+            }
+
+        }.build(appInfo)
+    }
+}
+```
+
+Notes:
+
+1. `Megazord.init()` is called before `createNimbus()`.
+1. `createNimbus` uses a `NimbusBuilder` to create the Nimbus object.
+1. Building a `JSONObject` of custom targeting attributes.
+1. The `nimbus.fetchExperiments()` method is called sometime at or after the app has started.
+
+## `NimbusBuilder` configuration
+
+### Getting errors out of `Nimbus`
+
+By design, Nimbus is deliberately unobtrusive; if it fails then it should not crash, but continue as if not enrolled in any experiments.
+
+The `errorReporter` callback is there to connect `Nimbus` to any error reporting framework in the rest of the app.
+
+```kotlin
+        return NimbusBuilder(context).apply {
+            // …
+            errorReporter = { message, e ->
+                Logger.error("Nimbus error: $message", e)
+            }
+            // …
+        }.build(appInfo)
+```
+
+### Connecting the `NimbusApi` to FML generated code
+
+The FML generated code has a runtime dependency on the `NimbusApi`.
+
+Two callbacks can be provided to the `NimbusBuilder`, one to connect the `NimbusApi` object when it is created,
+and one to clear the caching in the generated code.
+
+```kotlin
+        // Use the Nimbus builder to build a NimbusApi object.
+        return NimbusBuilder(context).apply {
+            // …
+            // Connect FxNimbus to the Nimbus SDK.
+            onCreateCallback = { nimbus ->
+                FxNimbus.initialize { nimbus }
+            }
+            onApplyCallback = {
+                FxNimbus.invalidateCachedValues()
+            }
+            // …
+        }.build(appInfo)
+```
+
+Note:
+
+`onApplyCallback` is a syntactic sugar for
+
+```kotlin
+nimbus.register(object : NimbusInterface.Observer {
+    // do after every applyPendingExperiments()
+    onApplyCallback(nimbus)
+})
+```
+
+However, `onApplyCallback` in `NimbusBuilder` is guaranteed to be executed on the first `applyPendingExperiments()`, i.e. on every startup.
+
+### Handling First Run experiments
+
+Since `fetchExperiments` from the remote settings URL is slow, and we wish to be able have access to the Nimbus experimental configuration as early in start up as possible, Nimbus downloads and caches the experiment recipes on the `n`th run of the app and only applies them and makes them available to the app at the beginning of the _next_ i.e. the `(n + 1)`th run of ther app.
```

Comment:
```suggestion
Since `fetchExperiments` from the remote settings URL is slow, and we wish to be able have access to the Nimbus experimental configuration as early in start up as possible, Nimbus downloads and caches the experiment recipes on the `n`th run of the app and only applies them and makes them available to the app at the beginning of the _next_ i.e. the `(n + 1)`th run of the app.
```